pkg/kds/server/grpc.go

```diff
@@ -0,0 +1,64 @@
+package server
+
+import (
+	"fmt"
+	"net"
+
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	kds_config "github.com/Kong/kuma/pkg/config/kds"
+
+	"google.golang.org/grpc"
+
+	"github.com/Kong/kuma/pkg/core"
+	"github.com/Kong/kuma/pkg/core/runtime/component"
+)
+
+const grpcMaxConcurrentStreams = 1000000
+
+var (
+	grpcServerLog = core.Log.WithName("kds-server").WithName("grpc")
+)
+
+type grpcServer struct {
+	server Server
+	config kds_config.KumaDiscoveryServerConfig
+}
+
+var (
+	_ component.Component = &grpcServer{}
+)
+
+func (s *grpcServer) Start(stop <-chan struct{}) error {
+	var grpcOptions []grpc.ServerOption
+	grpcOptions = append(grpcOptions, grpc.MaxConcurrentStreams(grpcMaxConcurrentStreams))
+	grpcServer := grpc.NewServer(grpcOptions...)
+
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", s.config.GrpcPort))
```

Comment:
I guess that is not a problem for K8s, but how safe is to listen on "all" interfaces in Universal? Shall we have also IP configuration to be able to denote the specific IP:port where the service is running?

Reply:
That is probably a nit though, as we have most of our servers behaving the same way. We might want to handle this in a separate task and to go and fix all the servers. Don't bother with it here.